x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java

```diff
@@ -24,62 +24,60 @@
  */
 public enum DataType {
 
-    // @formatter:off
-    //             esType            jdbc type,          size,              defPrecision,dispSize, int,   rat,   docvals
-    NULL(          "null",           JDBCType.NULL,      0,                 0,                 0,  false, false, false),
-    UNSUPPORTED(                     JDBCType.OTHER,     0,                 0,                 0,  false, false, false),
-    BOOLEAN(       "boolean",        JDBCType.BOOLEAN,   1,                 1,                 1,  false, false, false),
-    BYTE(          "byte",           JDBCType.TINYINT,   Byte.BYTES,        3,                 5,  true,  false, true),
-    SHORT(         "short",          JDBCType.SMALLINT,  Short.BYTES,       5,                 6,  true,  false, true),
-    INTEGER(       "integer",        JDBCType.INTEGER,   Integer.BYTES,     10,                11, true,  false, true),
-    LONG(          "long",           JDBCType.BIGINT,    Long.BYTES,        19,                20, true,  false, true),
+    // esType jdbc type, size, defPrecision,dispSize, int, rat, docvals
```

Comment:
Obviously, this format style was intentional, to improve code readability of the different data types SQL defines which, due to their count, would have been more difficult to track/follow. If there are exception that can be made to the wide formatting, I would put this one on the list, if possible.

Reply:
Entertainingly, before commit 418f4a2 I had excluded this file from formatting, but was persuaded to remove the exclusion by @jasontedor and @jaymode.

Other than simply excluding a file from formatting, the only other way to control formatting per-file is to turn it off for portions of a file. The issue with that is that I'm using that feature for doc snippets since they have different formatting requirements to the rest of the code (most notably line length). So you can surround code with `// tag::no_format` and `// end::no_format` but then you can fall foul of Checkstyle, and we're not going to start adding Checkstyle exclusions for this.

So it comes to a question of what is more valuable regarding this file - consist, automatic formatting, (eventually) the same as the rest of the codebase, or special formatting that has to be manually maintained?

What does everyone think?

Reply:
CC @costin

Reply:
The purpose of formatting is to get consistency across our code base and increase readability.
The expectation that every class inside the code base is going to be strictly formatted and there will be no exception is ... unrealistic.
You mentioned the doc snippets as exceptions and I would add this class is another one.

While I dislike exceptions myself, the formatting decreases readability, something we've experienced in the past hence the tags to prevent formatting that inside the IDEs.
It seems to me this case exposes a shortcoming of the current formatting infra, which is to exclude through comments, blocks of code.
I'm not sure how difficult it would be to add such a thing but I can see it being necessary to do this across the board.

P.S. Going forward, I recommend applying appropriate labels when formatting so potential owners can become aware of it and raise concerns if they have any.

Reply:
The format in this file is already not consistent in the manually formatted lines; there are inconsistencies line to line once you get past the display size field; spacing between columns varies from 0 to many and I personally find the lack of whitespace between parameters hard to read. Then intervals themselves have a different format than the rest of the enum types.

Another item to consider in this case, is a point brought up by @jasontedor, which is that if a new value is added to the enum it could potentially lead to needing to reformat every line of the enum type definitions in order to keep the spacing in this file and we want diffs/lines touched to be as small as possible for reviewing.

In terms of exceptions, doc snippets are unique in that they have different requirements. I'd really like for us to avoid adding exceptions in various projects and files; once we start doing that, exceptions become more likely.

Reply:
The point to having chosen a style and explicitly not having exceptions is so that the code base is consistently formatted, and we don’t have debates and nitting in PRs anymore. Once we leave it to authors to decide the style, or decide on exceptions, we’re back to where we were: an inconsistent code base and bickering about style.

The goal of a line per constructor parameter when a constructor needs to wrap to multiple lines is so that the diff when adding a single parameter is blatantly obvious what has changed, at the call sites, and the method definition itself. Otherwise, diffs are harder to read. The same reasoning applies to methods (definitions, and invocations) so we should be doing the same there.

Reply:
@costin I do understand why `DataType.java` is currently laid out as it is, and in the first iteration of this PR, I excluded the file from formatting to preserve the layout. However, I feel that @jasontedor and @jaymode are correct in that we should be striving to limit the number of exceptions, and err on the side of consistency.

If you'd like to see the result of a full format of the SQL code, you can opt-in the project in the [projectPathsToFormat array in build.gradle](https://github.com/elastic/elasticsearch/blob/master/build.gradle#L110), and run `./gradlew spotlessApply`. See also PR #48908, which enabled formatting for the Enrich project.

For background on why the formatter is configured the way it is: mostly I was guided by what Prettier (prettier.io) does, whose strategy is broadly to try to fit a statement / declaration etc on a single line, but when it has to wrap something then each part goes on a separate line. I’m fairly sure that Prettier stems from Wadler’s algorithm from “A prettier printer” but I haven’t read the paper enough to confidently state that.

Overally I was going for consistency of output and behaviour. I've reformatted the entire codebase and picked over the result, and I've noticed that the code currently has a mix of pretty much every style you can imagine. It's a mess, stylistically speaking.

Reply:
@jaymode All the enum declarations use the column/block approach due to the number of parameters for readability.
It's best effort since the values and headers make formatting tricky across all enums hence why there are 3 blocks so it's easy to reason about between then without taking too much whitespace.  
This is not just styling issue, there's been quite some effort along the way to make sure the precision, size and displaySize are correct for the drivers.
I don't follow the "what if we add a new parameter" argument - there are so many variables (is it one digit, a boolean, a constant like ExtType) that it's worth considering if and when we get to it.
Further more, I'm confused on what formatting style is used - on one hand the enum declaration is too large as there's too much whitespace and needs compressing, on the other if the constructor/method invocations and declarations break the line limit, it suddenly blows to one argument per line?

That's quite a departure from the current style used in ES that's worth discussing internally.

Reply:
>That's quite a departure from the current style used in ES that's worth discussing internally.

It's not a departure when there isn't consistent formatting across the code base. Some use the line-per formatting, some don't. It's a mess.